apps/settings/js/modules/base/event_emitter.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * EventEmitter implements basic event emitting logics. It also provides
+ * function that augments the ability to other objects. For example:
+ *  
+ *   function Module() {
+ *     this.EventEmitter.call(this, [eventName1, eventName2]);
+ *   }
+ *   EventEmitter.augment(Module.prototype);
+ *
+ * @module EventEmitter
+ */
+define(function() {
+  'use strict';
+
+  /**
+   * @class EventEmitter
+   * @returns {EventEmitter}
+   */
+  function EventEmitter(eventNames) {
+    if (eventNames && eventNames.length) {
+      this._eventListeners = {};
+      eventNames.forEach((eventName) => {
+        this._eventListeners[eventName] = [];
+      });
+    } else {
+      throw new Error('EventEmitter: no valid registered events');
+    }
+  }
+
+  /**
+   * Emit a event of a specific type and a given value.
+   *
+   * @access private
+   * @memberOf EventEmitter.prototype
+   * @param {String} eventName
+   * @param {Object} value
+   */
+  EventEmitter.prototype._emitEvent = function(eventName, value) {
```

Comment:
The module augmented by `EventEmitter` is considered as `EventEmitter` itself. So if `_emitEvent` event is not exposed, it is private.